lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -245,40 +251,92 @@ class ConvertAtenAddSubOp : public OpConversionPattern<AtenOpT> {
           op, "Only floating-point or integer datatype legalization supported");
     }
 
+    Type rhsAlphaMulElemType;
+    if (outElemTy.isa<mlir::FloatType>()) {
+      rhsAlphaMulElemType = outElemTy;
+    } else {
+      // if output type is 64, input type should also be 64
+      rhsAlphaMulElemType = rewriter.getIntegerType(32);
+    }
+
+    // if right is scalar, rhgType==None, which need to be manually cast to
+    // TensorType else right is tensor, rhsType==tensor<i32/i64/f32>
     Value rhsAsTensor;
     if (!rhsType) {
-      if (failed(torchScalarToTosaTensor(rewriter, op, op.getOther(), rhsAsTensor,
-                                         outElemTy, {})))
+      if (failed(torchScalarToTosaTensor(rewriter, op, op.getOther(),
+                                         rhsAsTensor, rhsAlphaMulElemType, {})))
         return rewriter.notifyMatchFailure(
             op, "Currently only scalar constants are supported for "
                 "conversion in TOSA operation");
+    } else if (rhsType.getElementType() != rhsAlphaMulElemType) {
+      // right is tensor, rhsType == tensor<i32/i64/f32>
+      // right must be cast to same type as the alpha, so MulOp success
+      rhs = rewriter.create<tosa::CastOp>(
+          op->getLoc(),
+          RankedTensorType::get(rhsType.getShape(), rhsAlphaMulElemType), rhs);
+      // reinitialize left value type to tensor<i32/f32>
```

Comment:
You say left value, but then change the rhsType.

Reply:
Done